### DISCUSSION TITLE
next/image for dynamically sized images

### DISCUSSION THREAD

**Discussion by amuttsch** [vercel/next.js#18848] [Help] [answered]
Hi,

first thank you for providing the `<Image>` component, this will ease my development especially with lazy loading.

I want to display images that have varying aspect ratios in a fixed size container. Using 

```
<Image
    src={coverUrl}
    width={500}
    height={500}
/>
```

the image get stretched to the 1x1 aspect ratio. This is due to the style that is applied: 

```
width: 0px;
height: 0px;
min-width: 100%;
max-width: 100%;
min-height: 100%;
max-height: 100%;
```

For my purposes I want only the min and max height set. How can I archive this? Thank you!

**Comment by Timer** [comment, marked as answer]
https://github.com/vercel/next.js/blob/canary/examples/image-component/pages/layout-fill.js

As seen in the [Image Demo](https://image-component.nextjs.gallery/layout-fill)!

**Reply by amuttsch** [reply]
Thanks, that works for me.

Is the link to the demo anywhere in the documentation that I overlooked? I didn't know about this one. Maybe it's worth pointing this out in the documentation as I guess this is a common use case.

**Reply by nghiepnguyen520** [reply]
@Timer, I do with this example above but the images don't have to optimization. **Is it not right to intro about the Image component of Next will optimization?**

> Lighthouse checked
![Screen Shot 2020-12-22 at 11 05 13](https://user-images.githubusercontent.com/37394413/102847726-a2d17400-4445-11eb-9dfb-4529832d6721.png)

**Reply by hughgardiner** [reply]
The links above no longer resolve, are there updated docs somewhere?

**Reply by hughgardiner** [reply]
appying style `object-fit: contain;` to the Image component in addition to passing prop `fill={true}`ended up working for me

**Comment by amuttsch** [comment]
@Timer Would it be possible to also define a layout type `raw` that just returns the `<img>` tag without the styles applied? It should take an optional `width` prop that corresponds to the image size configuration. This would make custom styling much easier because we have full control over the layout and positioning.

Thanks!

**Comment by helloimalastair** [comment]
Can I use the Image component from a separate folder? When I use it in the Components folder, it says "Module not found: Can't resolve 'next/image'", and I can't figure out if it is the wrong folder, or something else.
